jobrunner/docker.py

```diff
@@ -278,10 +279,15 @@ def run(name, args, volume=None, env=None, allow_network_access=False, label=Non
     # This is in addition to the default LABEL which is always applied
     if label is not None:
         run_args.extend(["--label", label])
-    if env:
-        for key, value in env.items():
-            run_args.extend(["--env", f"{key}={value}"])
-    subprocess_run(run_args + args, check=True, capture_output=True)
+    # To avoid leaking the values into the command line arguments we set them
+    # in the evnironment and tell Docker to fetch them from there
+    if env is None:
+        env = {}
+    for key, value in env.items():
+        run_args.extend(["--env", key])
+    subprocess_run(
+        run_args + args, check=True, capture_output=True, env=dict(os.environ, **env)
```

Comment:
Hmm, not sure about passing all of os.environ in here, as that exposes e..g DB creds to the docker container?

Would env=env not work? Or perhaps a whitelist of env keys from os.environ?

Reply:
Oh, it only exposes them to docker, not to the action, my bad

Reply:
But still, what is the need for providing os.environ? Can we just get away with env?